Riot/Modules/SplitView/SplitViewCoordinator.swift

```diff
@@ -149,54 +159,195 @@ final class SplitViewCoordinator: NSObject, SplitViewCoordinatorType {
+        // If the primary root controller of a UISplitViewController is a UINavigationController,
+        // it's possible to have nested navigation controllers due to private property `_allowNestedNavigationControllers` set to true (https://blog.malcolmhall.com/2017/01/27/default-behaviour-of-uisplitviewcontroller-collapsesecondaryviewcontroller/).
+        // So in this case returning false here will push the `detailNavigationController` on top of the `primaryNavigationController`.
+        // Sample primary view stack:
+        // primaryNavigationController[
+        //   MasterTabBarController,
+        //   detailNavigationController[RoomViewController, RoomInfoListViewController]]
+        // Note that normally pushing a navigation controller on top of a navigation controller don't work.
+        return false
     }
 }
 
-/// MARK: - UINavigationControllerDelegate
+// MARK: - TabBarCoordinatorDelegate
 extension SplitViewCoordinator: TabBarCoordinatorDelegate {
     func tabBarCoordinatorDidCompleteAuthentication(_ coordinator: TabBarCoordinatorType) {
         self.delegate?.splitViewCoordinatorDidCompleteAuthentication(self)
     }
 }
 
-/// MARK: - SplitViewMasterPresentableDelegate
+// MARK: - SplitViewMasterPresentableDelegate
 extension SplitViewCoordinator: SplitViewMasterPresentableDelegate {
-    func splitViewMasterPresentable(_ presentable: Presentable, wantsToDisplay detailPresentable: Presentable) {
-        MXLog.debug("[SplitViewCoordinator] splitViewMasterPresentable: \(presentable) wantsToDisplay detailPresentable: \(detailPresentable)")
+    
+    var detailModules: [Presentable] {
+        return self.detailNavigationRouter?.modules ?? []
+    }
+    
+    func splitViewMasterPresentable(_ presentable: Presentable, wantsToReplaceDetailWith detailPresentable: Presentable, popCompletion: (() -> Void)?) {
+        MXLog.debug("[SplitViewCoordinator] splitViewMasterPresentable: \(presentable) wantsToReplaceDetailWith detailPresentable: \(detailPresentable)")
         
         guard let detailNavigationController = self.detailNavigationController else {
             MXLog.debug("[SplitViewCoordinator] splitViewMasterPresentable: Failed to display because detailNavigationController is nil")
             return
         }
         
-        detailNavigationController.viewControllers = [detailPresentable.toPresentable()]
+        let detailController = detailPresentable.toPresentable()
+        
+        // Reset the detail navigation controller with the given detail controller
+        self.detailNavigationRouter?.setRootModule(detailPresentable, popCompletion: popCompletion)
+        
+        // This will call first UISplitViewControllerDelegate method:  `splitViewController(_:showDetail:sender:)`, if implemented, to give the opportunity to customise `UISplitViewController.showDetailViewController(:sender:)` behavior.
```

Comment:
```suggestion
        // This will call first UISplitViewControllerDelegate method: `splitViewController(_:showDetail:sender:)`, if implemented, to give the opportunity to customise `UISplitViewController.showDetailViewController(:sender:)` behaviour.
```